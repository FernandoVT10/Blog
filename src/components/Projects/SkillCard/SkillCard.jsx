import Link from "next/link";

import "./SkillCard.scss";

export default ({ skill, openProjects }) => {
    return (
        <Link href={`?skill=${skill.name}`}>
            <div
            className="skill mb-3"
            onClick={openProjects(skill._id)}>
                <div className="skill__image-container">
                    <img
                    className="skill__image"
                    src={`/img/skills/${skill.image}`}
                    alt={skill.name} />

                    <span
                    className="skill__image-hover"
                    style={{background: skill.color}}></span>
                </div>
                
                <div className="skill__title" style={{background: skill.color}}>
                    { skill.name }
                </div>
            </div>
        </Link>
    );
};