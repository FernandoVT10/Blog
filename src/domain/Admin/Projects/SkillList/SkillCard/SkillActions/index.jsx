import SkillActions from "./SkillActions";

export default SkillActions;