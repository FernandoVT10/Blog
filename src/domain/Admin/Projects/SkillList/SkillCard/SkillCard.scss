.admin-skill-list-skill-card {
    display: block;

    &__title {
        padding: 20px 0;
    }

    &__title-active {
        color: var(--secondary);
    }

    &__project-list {
        display: none;

        &--active {
            display: block;
        }
    }

    &__skill-actions {
        float: right;
        margin-right: 10px;
    }

    &__loader {
        padding: 10px;
        display: flex;
        justify-content: center;
        color: var(--secondary);
    }

    &__not-found {
        padding: 10px;
        color: var(--secondary);
        font-weight: bold;
    }
}