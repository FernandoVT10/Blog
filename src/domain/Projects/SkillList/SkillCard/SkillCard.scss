.skill {
    width: 100%;
    cursor: pointer;

    &__image-container {
        position: relative;
    }

    &__image {
        width: 100%;
        height: 200px;
        object-fit: cover;
    }
    
    &__image-hover {
        position: absolute;
        content: "";
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        opacity: 0;
        transition: .3s;
    }
    
    &:hover &__image-hover {
        opacity: .5;
    }
    
    &__skill-name {
        color:var(--white);
    }
    
    &__title {
        padding: 5px;
        color: var(--white);
        font-weight: bold;
        font-size: 16px;
        border-radius: 5px;
    }
}