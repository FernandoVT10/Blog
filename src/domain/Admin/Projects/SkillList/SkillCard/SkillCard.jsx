import SkillActions from "./SkillActions";
import ProjectCardList from "./ProjectCardList";

import SkillInput from "../SkillInput";

import { useState } from "react";

import "./SkillCard.scss";

const SkillCard =  ({ skill, deleteSkill, updateSkillName }) => {
    const [skillName, setSkillName] = useState(skill.name);
    const [isActive, setIsActive] = useState(false);
    const [isEditing, setIsEditing] = useState(false);

    const handleForm = e => {
        e.preventDefault();

        updateSkillName(skill._id, skillName);
        setIsEditing(false);
    }

    const handleOnBlur = () => {
        setIsEditing(false);
        setSkillName(skill.name);
    }

    const iconClass = isActive ? "fa-folder-open" : "fa-folder";
    const titleClass = isActive ? "admin-skill-list-skill-card__title-active" : "";
    const projectListClass = isActive ? "admin-skill-list-skill-card__project-list--active" : "";

    if(isEditing) {
        return (
            <SkillInput
            onSubmit={handleForm}
            skillName={skillName}
            setSkillName={setSkillName}
            onBlur={handleOnBlur}/>
        );
    }

    return (
        <div className="admin-skill-list-skill-card custom-table__body-row">
            <div className="admin-skill-list-skill-card__title">
                <span
                onClick={() => setIsActive(!isActive)}
                className={`custom-table__article__title ${titleClass}`}>
                    <i className={`fas mr-2 ${iconClass}`}></i>
                    { skill.name }
                </span>

                <div className="admin-skill-list-skill-card__skill-actions">
                    <SkillActions
                    skill={skill}
                    deleteSkill={deleteSkill}
                    setIsEditing={setIsEditing}/>
                </div>
            </div>

            <div className={`admin-skill-list-skill-card__project-list ${projectListClass}`}>
                <ProjectCardList skillName={skill.name} isActive={isActive}/>
            </div>
        </div>
    );
}

export default SkillCard;