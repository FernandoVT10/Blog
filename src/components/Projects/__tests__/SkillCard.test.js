import SkillCard from "../SkillCard/";
import { create } from "react-test-renderer";

const SKILL_MOCK = {
    _id: 18,
    name: "React JS",
    image: "skill-1.jpg",
    color: "blue"
};

test("Check if <SkillCard/> renders correctly", () => {
    const component = create(<SkillCard skill={SKILL_MOCK} openProjects={() => {}}/>);

    expect(component.toJSON()).toMatchSnapshot();
});