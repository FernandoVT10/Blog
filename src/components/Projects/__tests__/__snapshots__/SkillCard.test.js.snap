// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <SkillCard/> renders correctly 1`] = `
<div
  className="skill mb-3"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="skill__image-container"
  >
    <img
      alt="React JS"
      className="skill__image"
      src="/img/skills/skill-1.jpg"
    />
    <span
      className="skill__image-hover"
      style={
        Object {
          "background": "blue",
        }
      }
    />
  </div>
  <div
    className="skill__title"
    style={
      Object {
        "background": "blue",
      }
    }
  >
    React JS
  </div>
</div>
`;
